js/classes/entities/midground/weapon/bomb.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * A Bomb is a type of projectile that explodes after a period of time, dealing damage to nearby entities.
+ * 
+ * @author Nathan Hinthorne
+ */
+class Bomb {
+    /**
+     * @param {Vector} startPos The starting position of the Bomb.
+     * @param {Vector} targetPos The target position of the Bomb.
+     */
+    constructor(startPos, targetPos) {
+        this.base = new ProjectileBase(
+            this,
+            startPos,
+            targetPos,
+            Bomb.SCALED_SIZE,
+            Bomb.INITIAL_SPEED,
+            Bomb.WEIGHT,
+            (block) => this.onBlockCollision(block),
+            (enemy) => this.onEnemyCollision(enemy)
+        );
+
+        this.animations = [];
+        this.loadAnimations();
+
+        // this.numBounces = 0;
+
+        // this.bounceTimer = 0;
```

Comment:
Why did you leave these?

Reply:
I left some commented out bouncing functionality in case we need to swap it back in. As the game development is progressing, it's becoming clear I no longer need that code. I'll remove it.